File: qe-selenium/qe-selenium-python/Activity15.py
from selenium import webdriver
from selenium.webdriver import Firefox
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-attributes/")
    action=ActionChains(driver)

    print(driver.title)
    driver.find_element(By.XPATH,"//input[starts-with(@type,'text')]").click()
    action.send_keys("Pavan").perform()
    driver.find_element(By.XPATH,"//input[starts-with(@type,'email')]").click()
    action.send_keys("Pavan@g").perform()
    driver.find_element(By.XPATH,"//input[starts-with(@type,'date')]").send_keys("2002-12-12")
    driver.find_element(By.XPATH,"//textarea[contains(@id,'additional-details')]").click()
    action.send_keys("Pavan@g").perform()
    
    driver.find_element(By.XPATH,"//*[@id=\"signInForm\"]/button").click()
    print(driver.find_element(By.XPATH,"//*[@id=\"action-confirmation\"]").text)

File: qe-selenium/qe-selenium-java/src/main/java/com/demos/Activity15.java
package com.demos;

import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity15 {
    public static void main(String[] args) {
        FirefoxDriver driver =new FirefoxDriver();
        Actions actions = new Actions(driver);
        driver.get("https://training-support.net/webelements/dynamic-attributes/");
        System.out.println(driver.getTitle());

        driver.findElement(By.xpath("//input[starts-with(@type,'text')]")).sendKeys("pavan");
        driver.findElement(By.xpath("//input[starts-with(@type,'email')]")).sendKeys("pavan@gmail.com");
        driver.findElement(By.xpath("//input[@type='date']")).sendKeys("2026-04-22");
        driver.findElement(By.xpath("//textarea[contains(@id,'-additional-details-')]")).sendKeys("1221-12-12");



        // driver.findElement(By.xpath("//*[@id=\"full-name-6094\"]")).sendKeys("pavan");
        // driver.findElement(By.xpath("//*[@id=\"d177-email\"]")).sendKeys("pavan@gmail.com");
        // driver.findElement(By.xpath("//*[@id=\"signInForm\"]/input[3]")).sendKeys("1221-12-12");
        // WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));




    }
}   
